vhost_rs/src/vhost_user/slave_fs_cache.rs

```diff
@@ -9,6 +9,16 @@ use std::os::unix::io::RawFd;
 use std::os::unix::net::UnixStream;
 use std::sync::{Arc, Mutex};
 
+/// Trait for virtio-fs cache requests operations.  This is mainly used to hide
+/// vhost-user details from virtio-fs's fuse part.
+pub trait FsCacheReqHandler: Send + Sync + 'static {
+    /// Setupmapping operation
+    fn map(&mut self, fs: &VhostUserFSSlaveMsg, fd: RawFd) -> Result<()>;
+
+    /// Removemapping operation
+    fn unmap(&mut self, fs: &VhostUserFSSlaveMsg) -> Result<()>;
+}
```

Comment:
I don't think you should declare the trait here. Having the trait in `vhost_user_fs/src/fs_cache_req_handler.rs` is enough, and you can provide the implementation along with it directly in `vhost_user_fs` crate.

Reply:
I did it that way but then encountered the cyclic dependency problem (as I mentioned above [here](https://github.com/cloud-hypervisor/cloud-hypervisor/pull/698#discussion_r376673012) ).
Let me check if I can do a workaround.